### PR TITLE
TST: e2e on servicemap

### LINKED ISSUE #105: Use gRPC client instead of webclient
In the situp-core e2e test used gRPC to [export traces](https://github.com/opendistro-for-elasticsearch/simple-ingest-transformation-utility-pipeline/blob/a736dfc0caca13bc4fa46ce60380d519ddf51123/situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java#L99)

### PR DESCRIPTION
**Pull request by chenqi0805**
*Issue #, if available:*
#85 #105 

*Description of changes:*
Adds e2e on servicemap, also fixes #105 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java [hunk 1]
```diff
@@ -167,38 +232,70 @@ public static ExportTraceServiceRequest getExportTraceServiceRequest(List<Resour
...
             spansList.add(
                     getResourceSpans(
-                            UUID.randomUUID().toString(),
-                            UUID.randomUUID().toString(),
-                            getRandomBytes(8),
-                            getRandomBytes(8),
-                            getRandomBytes(16),
-                            SPAN_KINDS.get(RANDOM.nextInt(SPAN_KINDS.size()))
+                            serviceNames.get(i % serviceNames.size()),
+                            traceGroups.get(i % traceGroups.size()),
```
Comment (AustinTag) [inline situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:286]: Nit: Why move from random indexing to round robin indexing?

File: situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java [hunk 2]
```diff
@@ -70,41 +79,97 @@ public void testPipelineEndToEnd() throws IOException, InterruptedException {
 
         //Verify data in elasticsearch sink
         final List<Map<String, Object>> expectedDocuments = getExpectedDocuments(exportTraceServiceRequest1, exportTraceServiceRequest2);
+        final List<Map<String, Object>> expectedTargets = IntStream.range(0, serviceNames.size()).mapToObj(i -> {
```
Comment (AustinTag) [inline situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:82]: Wait, why are we doing this? This seems like a deterministic set of expected targets and edges for a random set of data?

File: situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java [hunk 3]
```diff
@@ -70,41 +79,97 @@ public void testPipelineEndToEnd() throws IOException, InterruptedException {
...
+            final Map<String, Object> target = new HashMap<>();
+            target.put("resource", serviceNames.get(i));
+            target.put("domain", traceGroups.get(i));
+            return target;
+        }).collect(Collectors.toList());
+        final List<Map<String, Object>> expectedEdges = IntStream.range(1, serviceNames.size() + 1).mapToObj(i -> {
+            final Map<String, Object> edge = new HashMap<>();
+            edge.put("traceGroupName", "tg1");
```
Comment (AustinTag) [inline situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:90]: We only expect tg1?

File: situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java (hunk 1 above)
Comment (AustinTag) [reply situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:286]: The reason I prefer randomly generating the data and inferring which edges we should see if that its gives us much better actual coverage over time. This type of generation only tells us that the service map gives us a more narrow set of coverage.

Comment (chenqi0805) [reply situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:286]: > The reason I prefer randomly generating the data and inferring which edges we should see if that its gives us much better actual coverage over time.

Agreed. The difficulty is how could we make assertions on edges while they are random in number and also in the "vertices"?

Comment (AustinTag) [reply situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:286]: Well you'd need to infer based on the data that you generate. If this works, it's not a blocker for me for now though.

Comment (chenqi0805) [reply situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndIntegrationTest.java:286]: Taking another dive.

File: situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndServiceMapTest.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            Span.SpanKind.SPAN_KIND_SERVER, Span.SpanKind.SPAN_KIND_INTERNAL, Span.SpanKind.SPAN_KIND_CLIENT, Span.SpanKind.SPAN_KIND_SERVER,
+            Span.SpanKind.SPAN_KIND_INTERNAL, Span.SpanKind.SPAN_KIND_CLIENT, Span.SpanKind.SPAN_KIND_SERVER);
+    private static final List<String> serviceNames2 = Arrays.asList("ServiceA", "ServiceA", "ServiceB", "ServiceA", "ServiceE");
+    private static final List<String> spanIds2 = Arrays.asList("101", "201", "301", "401", "501");
+    private static final List<String> parentIds2 = Arrays.asList(null, "101", "201", "101", "401");
+    private static final List<String> spanNames2 = Arrays.asList("VEGGIES", "CALL_SERVICE_B_ONION", "/ONION", "CALL_SERVICE_E_POTATO", "/POTATO");
+    private static final List<Span.SpanKind> spanKinds2 = Arrays.asList(
+            Span.SpanKind.SPAN_KIND_INTERNAL, Span.SpanKind.SPAN_KIND_CLIENT, Span.SpanKind.SPAN_KIND_SERVER, Span.SpanKind.SPAN_KIND_CLIENT,
+            Span.SpanKind.SPAN_KIND_SERVER);
+    private static final String SERVICE_MAP_INDEX_NAME = "otel-v1-apm-service-map";
```
Comment (kowshikn) [inline situp-core/src/integrationTest/java/com/amazon/situp/integration/EndToEndServiceMapTest.java:61]: nit: You could create a class or enum and manage these, the reason is we will be adding more metadata in the next versions.

```
public enum ServiceMapData {

    DATA_100("100", "serviceA", null, "FRUITS", Span.SpanKind.SPAN_KIND_INTERNAL);


    private final String spanId;
    private final String serviceName;
    private final String parentSpanId;
    private final String name;
    private final Span.SpanKind spanKind;

    ServiceMapData(String spanId, String serviceName, String parentSpanId, String name, Span.SpanKind kind) {
        this.spanId = spanId;
        this.serviceName = serviceName;
        this.parentSpanId = parentSpanId;
        this.name = name;
        this.spanKind = kind;
    }
}
```

**Review by kowshikn — APPROVED** [review]

**Review by AustinTag — APPROVED** [review]

**Pull request merged**